### PR TITLE
codeowners: update all obs links to use `obs-prs`

### PR DESCRIPTION
**Pull request by dhartunian**
Release note: None

### CODE REVIEW

**Review by abarganier — APPROVED** [review]
:lgtm: once comments fixed

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/121693#-:-NuZY3xw-xNPwN5pJE5N:b-kxfymg)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @dhartunian, @herkolategan, @jlinder, @nkodali, @srosenberg, and @xinhaoz)

___
*[`TEAMS.yaml` line 91 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/121693#-NuZXzzt697-k-tLkTfN:-NuZXzzt697-k-tLkTfO:b-4mz2u6) ([raw file](https://github.com/cockroachdb/cockroach/blob/097c882e63151ed14477c1c02ba651f89d8cab42/TEAMS.yaml#L91)):*
> ```YAML
>     cockroachdb/server-prs: other
>   triage_column_id: 2521812
> cockroachdb/admin-ui:
> ```

I think there's still an `admin-ui-prs` link lingering somewhere that's failing CI. 



_Code quote:_
```YAML
cockroachdb/admin-ui:
```
___
*[`TEAMS.yaml` line 96 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/121693#-NuZY9h5EZeClhgZgDq-:-NuZY9h6CxwjMYPvyKii:b-hukmd0) ([raw file](https://github.com/cockroachdb/cockroach/blob/097c882e63151ed14477c1c02ba651f89d8cab42/TEAMS.yaml#L96)):*
> ```YAML
>   triage_column_id: 6598672
>   label: T-observability-inf
> cockroachdb/obs-inf-prs:
> ```

Seems like we need to update `registry.OwnerObsInf` related code as well. 



_Code quote:_
```YAML
cockroachdb/obs-inf-prs:
```

File: pkg/cmd/roachtest/registry/owners.go [hunk 1]
```diff
@@ -29,15 +29,14 @@ const (
 	OwnerKV               Owner = `kv`
 	OwnerReplication      Owner = `replication`
 	OwnerAdmissionControl Owner = `admission-control`
-	OwnerObsInf           Owner = `obs-inf-prs`
```
Comment (jlinder) [inline pkg/cmd/roachtest/registry/owners.go:32]: I see this as used in a few other files too. I see it in:

* pkg/cmd/roachtest/tests/acceptance.go
* pkg/cmd/roachtest/tests/kv.go

**Review by jlinder — CHANGES_REQUESTED** [review]
A few changes are needed.

**Review by dhartunian — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/121693)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @abarganier, @herkolategan, @jlinder, @nkodali, @srosenberg, and @xinhaoz)

___
*[`TEAMS.yaml` line 91 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/121693#-NuZXzzt697-k-tLkTfN:-Nu_WheQ48TSKqTGDVRF:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/097c882e63151ed14477c1c02ba651f89d8cab42/TEAMS.yaml#L91)):*
<details><summary><i>Previously, abarganier (Alex Barganier) wrote…</i></summary><blockquote>

I think there's still an `admin-ui-prs` link lingering somewhere that's failing CI.
</blockquote></details>

Done.
___
*[`TEAMS.yaml` line 91 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/121693#gh-1549962733:-Nu_XIyi9cRN0P3j7hwF:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/e3c3e9d3c2fbd0d9f8c98be199be4eb1ee25935e/TEAMS.yaml#L91)):*
<details><summary><i>Previously, jlinder (James H. Linder) wrote…</i></summary><blockquote>

Because the observability team doesn't have a GitHub project.

```suggestion
  # The observability team uses Jira for managing issues. So there is no triage column ID.
```
</blockquote></details>

Done.
___
*[`TEAMS.yaml` line 96 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/121693#-NuZY9h5EZeClhgZgDq-:-Nu_Whwc964tJZ3HjA8m:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/097c882e63151ed14477c1c02ba651f89d8cab42/TEAMS.yaml#L96)):*
<details><summary><i>Previously, abarganier (Alex Barganier) wrote…</i></summary><blockquote>

Seems like we need to update `registry.OwnerObsInf` related code as well.
</blockquote></details>

Done.

**Review by jlinder — APPROVED** [review]
LGTM.

**Comment by dhartunian** [comment]
TFTRs

bors=jlinder,abarganier

**Comment by dhartunian** [comment]
bors r=jlinder,abarganier

**Pull request merged**